src/functions/edi/outbound/handler.ts

```diff
@@ -288,3 +285,53 @@ const filterDestination = (
 
   return true;
 };
+
+const prepapreEvent = async (event: unknown): Promise<OutboundEvent> => {
+  // check if we have a legacy event input
+  const legacyEventParse = LegacyOutboundEventSchema.safeParse(event);
+  if (legacyEventParse.success) {
+    // check if we can resolve the partnership using the legacy event
+    let partnership: NoUndefined<GetX12PartnershipCommandOutput> | undefined;
+    try {
+      partnership = await loadPartnershipById({
+        partnershipId: `${legacyEventParse.data.metadata.sendingPartnerId}_${legacyEventParse.data.metadata.receivingPartnerId}`,
+      });
+    } catch (error) {
+      // swallow error
+    }
+
+    if (partnership === undefined) {
+      try {
+        partnership = await loadPartnershipById({
+          partnershipId: `${legacyEventParse.data.metadata.receivingPartnerId}_${legacyEventParse.data.metadata.sendingPartnerId}`,
+        });
+      } catch (error) {
+        // swallow error
+      }
+    }
+
+    if (partnership === undefined)
+      throw new ErrorWithContext(
+        "Legacy input used for edi-outbound, but no partnership found",
+        legacyEventParse.data
+      );
+
+    // reshape legacy event to new event format
+    event = {
+      metadata: {
+        partnershipId: partnership.partnershipId,
+        usageIndicatorCode: "P", // Do we default to P?
```

Comment:
Legacy input doesn't include `usageIndicatorCode` - that used to be part of the old partnership record, but it's part of Partners API. Is hard coding to `P` the best call? Seems like the safest, it won't break a production setup, and if we're migrating a test setup they generally move the latest input format.